### ISSUE TITLE
Desktop: System.IO.MemoryMappedFiles.Tests.MemoryMappedFileTests_CreateFromFile.ValidArgumentCombinationsWithPath_ModesCreateOrTruncate failed with "System.ArgumentException"

### ISSUE THREAD

**Issue by Jiayili1** [dotnet/corefx#17726]
Configuration: outerloop_netfx_windows_nt_debug ([build#1](https://ci.dot.net/job/dotnet_corefx/job/master/job/outerloop_netfx_windows_nt_debug/1/testReport/System.IO.MemoryMappedFiles.Tests/MemoryMappedFileTests_CreateFromFile/FileDoesNotExist_mode__Truncate_/))

Failed test: System.IO.MemoryMappedFiles.Tests.MemoryMappedFileTests_CreateFromFile.ValidArgumentCombinationsWithPath_ModesCreateOrTruncate

Message:
~~~
System.ArgumentException : Combining FileMode: Truncate with FileSystemRights: ReadData, CreateFiles is invalid. FileMode.Truncate is valid only when used with FileSystemRights.Write.
~~~

Stack Trace:
~~~
at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost) at System.IO.FileStream..ctor(String path, FileMode mode, FileSystemRights rights, FileShare share, Int32 bufferSize, FileOptions options) at System.IO.MemoryMappedFiles.MemoryMappedFile.CreateFromFile(String path, FileMode mode, String mapName, Int64 capacity, MemoryMappedFileAccess access) at System.IO.MemoryMappedFiles.MemoryMappedFile.CreateFromFile(String path, FileMode mode, String mapName, Int64 capacity) at System.IO.MemoryMappedFiles.Tests.MemoryMappedFileTests_CreateFromFile.ValidArgumentCombinationsWithPath_ModesCreateOrTruncate(FileMode mode, String mapName, Int64 capacity, MemoryMappedFileAccess access) in D:\j\workspace\outerloop_net---903ddde6\src\System.IO.MemoryMappedFiles\tests\MemoryMappedFile.CreateFromFile.Tests.cs:line 368
~~~

Detail: https://ci.dot.net/job/dotnet_corefx/job/master/job/outerloop_netfx_windows_nt_debug/1/testReport/System.IO.MemoryMappedFiles.Tests/MemoryMappedFileTests_CreateFromFile/ValidArgumentCombinationsWithPath_ModesCreateOrTruncate_mode__Truncate__mapName__null__capacity__1__access__ReadWrite_/

**Comment by danmosemsft** [comment]
@ianhays  do you mind taking this one

**Comment by safern** [comment]
I don't see this one failing anymore. Seems to be fixed on PR: #17625
